edge/yaml/yaml.go

```diff
@@ -34,7 +35,11 @@ func NewYAML(fileContent string, credentials []agent.RegistryCredentials) *yaml
 func (y *yaml) getRegistryCredentialsByImageURL(imageURL string) []agent.RegistryCredentials {
 	credentials := []agent.RegistryCredentials{}
 	for _, r := range y.registryCredentials {
-		if strings.Contains(imageURL, r.ServerURL) {
+		domain, _, err := getRegistryAndPath(imageURL)
```

Comment:
Any reason why this returns a `path` if we just throw it out and don't use this function anywhere else? Are you thinking you may want the path in the future?